### PR TITLE
MM-16712 Allow specifying number of minio and mysql replicas in ClusterInstallation spec

### PR DESCRIPTION
**Pull request by jwilander**
https://mattermost.atlassian.net/browse/MM-16712

### CODE REVIEW

File: pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go [hunk 1]
```diff
@@ -65,7 +72,11 @@ type DatabaseType struct {
 	// Defines the storage size for the database. ie 50Gi
 	// +optional
 	// +kubebuilder:validation:Pattern=^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$
-	DbStorageSize string `json:"dbStorageSize,omitempty"`
+	DatabaseStorageSize string `json:"databaseStorageSize,omitempty"`
```
Comment (jwilander) [inline pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go]: Updated to match the naming conventions of the other fields

File: docs/minio-operator/minio-operator.yaml [hunk 2]
```diff
@@ -87,6 +87,6 @@ spec:
       serviceAccountName: minio-operator-sa
       containers:
       - name: minio-operator
-        image: minio/k8s-operator:latest
+        image: mattermost/minio-operator
```
Comment (jwilander) [inline docs/minio-operator/minio-operator.yaml]: Pointing at our Docker image to support standalone Minio until https://github.com/minio/minio-operator/pull/23 is accepted and also until the Minio operator is tagged so we don't have to worry about breaking changes

**Comment by gabrieljackson** [comment]
Will wait to review until tests pass in case a change is required.

**Comment by jwilander** [comment]
@cpanato I'm unsure if the tests are failing because of my changes or if the CI machine is just running out of memory, any idea?

**Comment by cpanato** [comment]
@jwilander checking

**Comment by cpanato** [comment]
@jwilander the issue is in the tests you are waiting for 1 deployment for minio and mysql but both have 4 pods
```
•100% ➜ kubectl get pod -n mattermost-operator
NAME                                  READY   STATUS     RESTARTS   AGE
mattermost-operator-8dcb8db6f-hlh9x   1/1     Running    0          7m48s
test-mm-56965bf9cf-nskrq              1/1     Running    0          7m32s
test-mm-minio-0                       1/1     Running    0          7m33s
test-mm-minio-1                       1/1     Running    0          7m17s
test-mm-minio-2                       1/1     Running    0          7m4s
test-mm-minio-3                       1/1     Running    0          6m24s
test-mm-mysql-mysql-0                 4/4     Running    0          7m31s
test-mm-mysql-mysql-1                 4/4     Running    0          5m58s
test-mm-mysql-mysql-2                 4/4     Running    0          3m33s
test-mm-mysql-mysql-3                 0/4     Init:1/2   0          66s
```

**Comment by cpanato** [comment]
also in the test you are waiting for deployment and both minio and mysql are not deployments they are statefullsets :)

File: pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go (hunk 1 above)
Comment (cpanato) [reply pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go]: 👍

File: pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go [hunk 3]
```diff
@@ -65,7 +72,11 @@ type DatabaseType struct {
...
+	// DatabaseReplicas defines the number of database replicas
+	// For redundancy use at least 2 replicas
```
Comment (gabrieljackson) [inline pkg/apis/mattermost/v1alpha1/clusterinstallation_types.go]: nit:
```suggestion
	// For redundancy use at least 2 replicas.
```

**Comment by jwilander** [comment]
Thanks for the test help @cpanato !

**Pull request merged**